docs/developer-guide/intro.md

```diff
@@ -0,0 +1,4 @@
+---
```

Comment:
What are we suppose to put in this file?

Reply:
@alijnmerchant21 a basic introduction, it's WIP, adding more commits to review :)

Reply:
The first section of content in the guide: most likely, a brief intro to Starport.